lib/resourcemerge/core.go

```diff
@@ -466,12 +466,10 @@ func ensureVolumeSourceDefaults(required *corev1.VolumeSource) {
 }
 
 func ensureSecurityContextPtr(modified *bool, existing **corev1.SecurityContext, required *corev1.SecurityContext) {
-	// if we have no required, then we don't care what someone else has set
-	if required == nil {
+	if *existing == nil && required == nil {
 		return
 	}
-
-	if *existing == nil {
+	if *existing == nil || (required == nil && *existing != nil) {
```

Comment:
How does it handle `existing != nil , required != nil  but existing != require ` case? 
Can we add code comments about which cases the conditional loops are handling.